cf_units/__init__.py

```diff
@@ -1842,10 +1842,12 @@ def convert(self, value, other, ctype=FLOAT64, inplace=False):
                 ut1 = self.utime()
                 ut2 = other.utime()
                 result = ut2.date2num(ut1.num2date(result))
-                # Preserve the datatype of the input array if it was float32.
-                if (isinstance(value, np.ndarray) and
-                   value.dtype == np.float32):
-                    result = result.astype(np.float32)
+                # Preserve the datatype of the input array if it was float.
+                if isinstance(value, np.ndarray):
+                    if value.dtype == np.float32:
+                        result = result.astype(np.float32)
+                    elif value.dtype == np.float64:
+                        result = result.astype(np.float64)
```

Comment:
@bjlittle are you OK with these changes? I have to confess I'm not sure why that was there in the first place. Maybe we should just always convert to the original dtype?

Reply:
Here is the original addition: https://github.com/SciTools/iris/pull/1385

So it was about making the time coordinate consistent for all cubes in a cubelist.  So it seems to me that we should also preserve float64 (but I have no authority here!)

I think trying to preserve ints could be problematic: if you convert from say “hours since ...” to “days since...” and not everything is midnight, the output would need to be float.

Reply:
Thanks for your wisdom @rcomer, I agree. Most of the original cf-units devs have moved on so we all have about as much authority as each other here!

Reply:
@trexfeathers when I wrote that comment I hadn't yet contributed in `cf-units`, so I probably have more authority now than I did then!